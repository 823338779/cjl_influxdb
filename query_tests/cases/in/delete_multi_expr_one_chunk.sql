-- Demonstrate soft deleted rows will not be return to queries
-- IOX_SETUP: OneDeleteMultiExprsOneChunk

-- select *
SELECT * from cpu;

SELECT time, bar from cpu;

SELECT bar from cpu;

SELECT count(time), count(*), count(bar), min(bar), max(bar), min(time), max(time)  from cpu;

SELECT count(time)  from cpu;

SELECT count(foo) from cpu;

SELECT count(bar) from cpu;

SELECT count(*) from cpu;

SELECT min(bar) from cpu;

SELECT foo from cpu;

SELECT min(foo) from cpu;
SELECT max(foo) from cpu;

SELECT min(foo) as min_foo from cpu group by time order by min_foo;
SELECT max(foo) as max_foo from cpu group by time order by max_foo;
SELECT time, max(foo) as max_foo from cpu group by time order by time, max_foo;

SELECT min(foo) as min_foo from cpu group by bar order by min_foo;
SELECT bar, max(foo) as max_foo from cpu group by bar order by bar, max_foo;
SELECT max(foo) as max_foo from cpu group by time order by max_foo; 

SELECT min(time) from cpu;
SELECT max(time) from cpu;

SELECT min(time) from cpu group by bar;
SELECT bar, min(time) as min_time from cpu group by bar order by bar, min_time;
SELECT max(time) as max_time from cpu group by foo order by max_time;
SELECT foo, max(time) as max_time from cpu group by foo order by foo, max_time;


SELECT time from cpu;

SELECT max(bar) from cpu;

--------------------------------------------------------
-- With selection predicate

SELECT * from cpu where bar >= 1.0 order by bar, foo, time;

SELECT foo from cpu where bar >= 1.0 order by foo;

SELECT time, bar from cpu where bar >= 1.0 order by bar, time;

SELECT * from cpu where foo = 'you' order by bar, foo, time;

SELECT min(bar) as mi, max(time) as ma from cpu where foo = 'you' order by mi, ma


