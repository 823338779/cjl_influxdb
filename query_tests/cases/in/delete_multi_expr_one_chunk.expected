-- Test Setup: OneDeleteMultiExprsOneChunk
-- SQL: SELECT * from cpu;
+-----+-----+--------------------------------+
| bar | foo | time                           |
+-----+-----+--------------------------------+
| 1   | me  | 1970-01-01T00:00:00.000000040Z |
| 2   | you | 1970-01-01T00:00:00.000000020Z |
+-----+-----+--------------------------------+
-- SQL: SELECT time, bar from cpu;
+--------------------------------+-----+
| time                           | bar |
+--------------------------------+-----+
| 1970-01-01T00:00:00.000000040Z | 1   |
| 1970-01-01T00:00:00.000000020Z | 2   |
+--------------------------------+-----+
-- SQL: SELECT bar from cpu;
+-----+
| bar |
+-----+
| 1   |
| 2   |
+-----+
-- SQL: SELECT count(time), count(*), count(bar), min(bar), max(bar), min(time), max(time)  from cpu;
+-----------------+-----------------+----------------+--------------+--------------+--------------------------------+--------------------------------+
| COUNT(cpu.time) | COUNT(UInt8(1)) | COUNT(cpu.bar) | MIN(cpu.bar) | MAX(cpu.bar) | MIN(cpu.time)                  | MAX(cpu.time)                  |
+-----------------+-----------------+----------------+--------------+--------------+--------------------------------+--------------------------------+
| 2               | 2               | 2              | 1            | 2            | 1970-01-01T00:00:00.000000020Z | 1970-01-01T00:00:00.000000040Z |
+-----------------+-----------------+----------------+--------------+--------------+--------------------------------+--------------------------------+
-- SQL: SELECT count(time)  from cpu;
+-----------------+
| COUNT(cpu.time) |
+-----------------+
| 2               |
+-----------------+
-- SQL: SELECT count(foo) from cpu;
+----------------+
| COUNT(cpu.foo) |
+----------------+
| 2              |
+----------------+
-- SQL: SELECT count(bar) from cpu;
+----------------+
| COUNT(cpu.bar) |
+----------------+
| 2              |
+----------------+
-- SQL: SELECT count(*) from cpu;
+-----------------+
| COUNT(UInt8(1)) |
+-----------------+
| 2               |
+-----------------+
-- SQL: SELECT min(bar) from cpu;
+--------------+
| MIN(cpu.bar) |
+--------------+
| 1            |
+--------------+
-- SQL: SELECT foo from cpu;
+-----+
| foo |
+-----+
| me  |
| you |
+-----+
-- SQL: SELECT min(foo) from cpu;
+--------------+
| MIN(cpu.foo) |
+--------------+
| me           |
+--------------+
-- SQL: SELECT max(foo) from cpu;
+--------------+
| MAX(cpu.foo) |
+--------------+
| you          |
+--------------+
-- SQL: SELECT time from cpu;
+--------------------------------+
| time                           |
+--------------------------------+
| 1970-01-01T00:00:00.000000040Z |
| 1970-01-01T00:00:00.000000020Z |
+--------------------------------+
-- SQL: SELECT max(bar) from cpu;
+--------------+
| MAX(cpu.bar) |
+--------------+
| 2            |
+--------------+
-- SQL: SELECT * from cpu where bar >= 1.0 order by bar, foo, time;
+-----+-----+--------------------------------+
| bar | foo | time                           |
+-----+-----+--------------------------------+
| 1   | me  | 1970-01-01T00:00:00.000000040Z |
| 2   | you | 1970-01-01T00:00:00.000000020Z |
+-----+-----+--------------------------------+
-- SQL: SELECT foo from cpu where bar >= 1.0 order by foo;
+-----+
| foo |
+-----+
| me  |
| you |
+-----+
-- SQL: SELECT time, bar from cpu where bar >= 1.0 order by bar, time;
+--------------------------------+-----+
| time                           | bar |
+--------------------------------+-----+
| 1970-01-01T00:00:00.000000040Z | 1   |
| 1970-01-01T00:00:00.000000020Z | 2   |
+--------------------------------+-----+
-- SQL: SELECT * from cpu where foo = 'you' order by bar, foo, time;
+-----+-----+--------------------------------+
| bar | foo | time                           |
+-----+-----+--------------------------------+
| 2   | you | 1970-01-01T00:00:00.000000020Z |
+-----+-----+--------------------------------+
-- SQL: SELECT min(bar) as mi, max(time) as ma from cpu where foo = 'you' order by mi, ma
+----+--------------------------------+
| mi | ma                             |
+----+--------------------------------+
| 2  | 1970-01-01T00:00:00.000000020Z |
+----+--------------------------------+
